app/liquid_tags/github_tag/github_issue_tag.rb

```diff
@@ -46,9 +53,19 @@ def parse_link(link)
     end
 
     def generate_api_link(input)
-      input = input.gsub(/\?.*/, "")
-      input = input.gsub(/\d{1,}#issuecomment-/, "comments/") if input.include?("#issuecomment-")
-      "https://api.github.com/repos/#{input.gsub(/.*github\.com\//, '')}"
+      path = input.gsub(/\?.*/, "")
```

Comment:
Seeing this makes me think we should deal with an URI object here. Then we could interact with `query`, `fragment` etc. instead of doing this manually.

```ruby
uri = URI.parse(input)
uri.host = nil if uri.host == "github.com"
uri.path = uri.path.gsub(%r{/pull/}, "/pulls/")
uri.fragment = nil if uri.fragment.start_with?("#issue-")
# other transformation
uri.to_s
```

Reply:
make sense! I'll apply it!